### PR TITLE
unixtools: refactor for less redundancy

### PR DESCRIPTION
**Pull request by Synthetica9**
###### Motivation for this change

`unix-tools` didn't quite follow the DRY principle, refactored it to follow this principle better. 

This shouldn't actually change the output of any derivation, just the way we get there, so @GrahamcOfBorg shouldn't give any rebuilds. 

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/top-level/unix-tools.nix [hunk 1]
```diff
@@ -27,126 +26,121 @@ let
...
-  };
-  wall = singleBinary "wall" {
-    linux = pkgs.utillinux;
-  };
-  write = singleBinary "write" {
-    linux = pkgs.utillinux;
-    darwin = pkgs.darwin.basic_cmds;
+
+  makeCompat = name': value: buildEnv {
+      name = name' + "-compat";
```
Comment (matthewbauer) [inline pkgs/top-level/unix-tools.nix]: Make sure these have 2 space indentation.

**Review by matthewbauer — APPROVED** [review]
Looks good! Not huge line saving but will make it easier to add more.

File: pkgs/top-level/unix-tools.nix (hunk 1 above)
Comment (Synthetica9) [reply pkgs/top-level/unix-tools.nix]: :heavy_check_mark:

**Comment by Synthetica9** [comment]
(Looks like master machine broke, probably shouldn't merge this until that is fixed, just to make sure it won't trigger a mass rebuild)

**Comment by matthewbauer** [comment]
The eval error was hopefully unrelated.

@GrahamcOfBorg eval

**Comment by Synthetica9** [comment]
Looks like its fine now

**Comment by matthewbauer** [comment]
Thanks!

**Comment by dtzWill** [comment]
Unfortunately this complicates merging with staging, which made changes to unix-tools before the reformatting.

Can someone take a look? Thanks!

**Comment by matthewbauer** [comment]
Just merged master into staging.

**Comment by dtzWill** [comment]
@matthewbauer haha thanks! :smile_cat:

**Comment by Synthetica9** [comment]
Wew, that was quick :sweat_smile:

**Pull request merged**